kayenta/kayenta-influxdb/src/main/java/com/netflix/kayenta/influxdb/config/InfluxDbResponseConverter.java

```diff
@@ -50,80 +47,103 @@ public InfluxDbResponseConverter(ObjectMapper kayentaObjectMapper) {
-      for (int i = 1;
-          i < seriesColumns.size();
-          i++) { // Starting from index 1 to skip 'time' column
-
-        String id = seriesColumns.get(i);
-        long firstTimeMillis = extractTimeInMillis(seriesValues, 0);
-        long stepMillis = calculateStep(seriesValues, firstTimeMillis);
-        List<Double> values = new ArrayList<>(seriesValues.size());
-        for (List<Object> valueRow : seriesValues) {
-          if (valueRow.get(i) != null) {
-            String val = valueRow.get(i).toString();
-            values.add(Double.valueOf(val));
-          }
+        Map result = getResultObject(json);
+        List<Map> seriesList = (List<Map>) result.get("series");
+
+        if (CollectionUtils.isEmpty(seriesList)) {
+          log.warn("Received no data from InfluxDB.");
+          return null;
         }
-        influxDbResultsList.add(new InfluxDbResult(id, firstTimeMillis, stepMillis, null, values));
-      }
 
-      log.debug("Converted response: {} ", influxDbResultsList);
-      return influxDbResultsList;
-    } catch (IOException e) {
-      e.printStackTrace();
-    }
+        Map series = seriesList.get(0);
+        List<String> seriesColumns = (List<String>) series.get("columns");
+        List<List> seriesValues = (List<List>) series.get("values");
+        List<InfluxDbResult> influxDbResultsList = new ArrayList<>(seriesValues.size());
+
+        // TODO(joerajeev): if returning tags (other than the field names) we will need to skip tags
+        // from this loop,
+        // and to extract and set the tag values to the influxDb result.
+        for (int i = 1;
+            i < seriesColumns.size();
+            i++) { // Starting from index 1 to skip 'time' column
+          String id = seriesColumns.get(i);
+          long firstTimeMillis = extractTimeInMillis(seriesValues, 0);
+          long stepMillis = calculateStep(seriesValues, firstTimeMillis);
+          List<Double> values = new ArrayList<>(seriesValues.size());
+
+          for (List<Object> valueRow : seriesValues) {
+            if (valueRow.get(i) != null) {
+              String val = valueRow.get(i).toString();
+              values.add(Double.valueOf(val));
+            }
+          }
 
-    return null;
-  }
+          influxDbResultsList.add(
+              new InfluxDbResult(id, firstTimeMillis, stepMillis, null, values));
+        }
 
-  private Map getResultObject(String json)
-      throws IOException, JsonParseException, JsonMappingException, ConversionException {
-    Map responseMap = kayentaObjectMapper.readValue(json, Map.class);
-    List<Map> results = (List<Map>) responseMap.get("results");
-    if (CollectionUtils.isEmpty(results)) {
-      throw new ConversionException("Unexpected response from influxDb");
+        log.debug("Converted response: {}", influxDbResultsList);
+        return influxDbResultsList;
+      } catch (IOException e) {
+        e.printStackTrace();
+        throw e;
```

Comment:
The old behavior to swallow the exception and return null doesn't seem great, but I'm tempted to leave it unless there's some compelling reason to change it.